### PR TITLE
New checkbox style, use mixin to override checkboxes, some style updates too

### PR DESCRIPTION
**Pull request by nickgros**
A smattering of screenshots from various components:
image

image

image

image

image

### CODE REVIEW

File: src/lib/containers/EntityLink.tsx [hunk 1]
```diff
@@ -44,17 +44,19 @@ export const EntityLink = (props: EntityLinkProps) => {
             className={className}
             target="_blank"
             rel="noopener noreferrer"
-            href={`${PRODUCTION_ENDPOINT_CONFIG.PORTAL}#!Synapse:${entity.id!}${versionNumber ? `.${versionNumber}` : ''}`}
+            href={`${PRODUCTION_ENDPOINT_CONFIG.PORTAL}#!Synapse:${entity.id!}${
+              versionNumber ? `.${versionNumber}` : ''
+            }`}
           >
-            <EntityTypeIcon type={type} style={{ marginRight: '5px' }} />
+            <EntityTypeIcon type={type} style={{ marginRight: '6px' }} />
```
Comment (nickgros) [inline src/lib/containers/EntityLink.tsx:51]: Just giving this icon some space to breathe

File: src/lib/style/components/_markdown.scss [hunk 2]
```diff
@@ -59,7 +59,7 @@
   word-wrap: break-word;
   overflow-x: auto;
   overflow-y: hidden;
-  line-height: 1.8em;
+  line-height: 1.5em;
```
Comment (nickgros) [inline src/lib/style/components/_markdown.scss:62]: This large line height is what's causing [SWC-6031](https://sagebionetworks.jira.com/browse/SWC-6031). 1.5 is not much different and makes the linked issue much less noticable

File: src/lib/style/components/entity_finder/_entity-finder.scss [hunk 3]
```diff
@@ -7,16 +7,12 @@ $-header-height: 45px;
...
 $-selected-hover-background-color: color.adjust(
   $-selected-background-color,
-  $lightness: -5%
-);
-$-hover-background-color: color.change(
-  SRC.$primary-action-color-active,
-  $alpha: 0.2
+  $lightness: -10%
 );
+$-hover-background-color: map.get(SRC.$primary-color-palette, 100);
```
Comment (nickgros) [inline src/lib/style/components/entity_finder/_entity-finder.scss:15]: Style updates for the Entity Finder to match designs, and not conflict with the new checkbox

https://user-images.githubusercontent.com/17580037/162285895-0d433837-ad9e-46a1-a43f-9137cc6f46a7.mov

File: src/lib/style/components/entity_finder/_tree-node-browse.scss [hunk 4]
```diff
@@ -21,9 +21,7 @@
     }
   }
 
-  .Node.BrowseNode[aria-disabled='false'][aria-selected='true']
-    > .EntityName
-    > span {
+  .Node.BrowseNode[aria-disabled='false'][aria-selected='true'] > .EntityName {
     font-weight: bold;
   }
```
Comment (nickgros) [inline src/lib/style/components/entity_finder/_tree-node-browse.scss:26]: Last refactor of this component dropped the span but I didn't update the style.

File: src/lib/template_style/_form.scss [hunk 5]
```diff
@@ -19,6 +21,12 @@
   margin-top: 0px;
   margin-bottom: 0px;
 }
+
+.checkbox label,
+.radio label {
+  padding-left: 0px;
+}
+
```
Comment (nickgros) [inline src/lib/template_style/_form.scss:29]: This was just moved up

File: src/lib/template_style/_form.scss [hunk 6]
```diff
@@ -27,120 +35,156 @@
...
-  // We don't hide the input tag with display: none because it makes the input inaccessible (you can't 'tab' into it)
-  // See https://www.a11ywithlindsey.com/blog/create-custom-keyboard-accesible-checkboxes
-  position: absolute !important;
-  height: 1px;
-  width: 1px;
-  overflow: hidden;
-  clip: rect(1px, 1px, 1px, 1px);
-  margin-left: unset;
-}
+@mixin customCheckbox($checkbox-top-offset: 0px) {
```
Comment (nickgros) [inline src/lib/template_style/_form.scss:38]: All of the custom checkbox/radio styles now live in this mixin. Absolute positioning offsets are relative to each other. The total offset can be overridden by embedding this mixin with a custom parameter.

File: src/lib/template_style/_form.scss [hunk 7]
```diff
@@ -27,120 +35,157 @@
...
+  input[type='checkbox']:checked + span::after {
+    content: '';
+    transform: rotate(45deg);
+    height: 9px;
+    width: 4px;
+    border-bottom: 2px solid white;
+    border-right: 2px solid white;
+    position: absolute;
+    top: $checkbox-after-top-offset;
+    left: $checkbox-after-left-offset;
```
Comment (nickgros) [inline src/lib/template_style/_form.scss:166]: Change the checkmark from unicode character (appearance depends on OS/browser) to a custom-sized div and with two adjacent borders, which happens to end up looking like the checkmark in our designs.

**Review by conundrumgirl — APPROVED** [review]

**Pull request merged**